shell/src/main/java/tachyon/shell/command/CopyFromLocalCommand.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.shell.command;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.util.List;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.Lists;
+import com.google.common.io.Closer;
+
+import tachyon.Constants;
+import tachyon.TachyonURI;
+import tachyon.client.file.FileOutStream;
+import tachyon.client.file.TachyonFile;
+import tachyon.client.file.TachyonFileSystem;
+import tachyon.client.file.options.OutStreamOptions;
+import tachyon.conf.TachyonConf;
+import tachyon.exception.ExceptionMessage;
+import tachyon.exception.FileAlreadyExistsException;
+import tachyon.exception.TachyonException;
+import tachyon.shell.TfsShellUtils;
+import tachyon.thrift.FileInfo;
+import tachyon.util.io.PathUtils;
+
+/**
+ * Copy the specified file specified by "source path" to the path specified by "remote path".
+ * This command will fail if "remote path" already exists.
+ */
+public class CopyFromLocalCommand extends AbstractTfsShellCommand {
+
+  public CopyFromLocalCommand(TachyonConf conf, TachyonFileSystem tfs) {
+    super(conf, tfs);
+  }
+
+  @Override
+  public String getCommandName() {
+    return "copyFromLocal";
+  }
+
+  @Override
+  protected int getNumOfArgs() {
+    return 2;
```

Comment:
@yupeng9 I want to leave the methods in the interface but implement them at the abstract class level instead of within every individual command.

@siyangy Sorry, I meant the `super()` constructor. I'm suggesting that the abstract class own number of parameters and command name, and that it gets them through the subclass constructor. This shouldn't change anything about how the subclass constructor is called.